### ISSUE TITLE
Invoke component in a Vue plugin (outside the app)

### ISSUE THREAD

**Issue by MatanYadaev** [vuejs/composition-api#94]
I have toast component which I used to invoke using a Vue plugin, it worked fine with the class decorator API, but with this API I'm not sure how to make this work:

```ts
import { PluginObject } from 'vue';
import AppToast from '@/components/AppToast.vue';

const Toast: PluginObject<any>  = {
    install: (Vue) => {
        const node = document.createElement('div');
        document.body.appendChild(node);
        const toast = new AppToast();
        toast.$mount(node);

        Vue.prototype.$toast = toast;
        // Now `this.$toast.show('message here') is available from every component in the app
    },
};

export default Toast;
```

Any suggestions how to make this work?

**Comment by LinusBorg** [comment]
You can use `context.root.$toast` I think.

Sidenote: In general, I think that extending the prototype like that will not be encourages in Vue 3. While it's quick & easy it's not very clean. Plugins will probably be using provide/inject a lot more to inject stuff into an app.

**Comment by MatanYadaev** [comment]
@LinusBorg The issue with my example is that with this API you can't easily invoke the component as I did previously: `new AppToast()`
I tried also `AppToast.setup()` and `AppToast.render()`, but it didn't work.
There is a way to invoke a component from outside?

**Comment by LinusBorg** [comment]
Oh, I missed that part, thought this was about accessing `this.$toast` primarily.

Well you can totally still do that in Vue 2, provided that the .vue file actually exports a component (`Vue.extend({ ... })`) and not just an options object (`{ ... }`). This is totally unrelated to `setup` and the composition-api plugin, by the way. 

If it exports an options object, just do this:

```javascript
const toast = new Vue.extend(AppToast)();
```

It will work a bit differently in Vue 3, presumably.

**Comment by MatanYadaev** [comment]
@LinusBorg I see, cool, well I implemented it and I have new issue now:
This is my plugin:
```js
const Toast = {
    install: Vue => {
        const node = document.createElement('div');
        document.body.appendChild(node);
        const sdToast = new (Vue.extend(SdToast));
        sdToast.$mount(node);

        Vue.prototype.$toast = sdToast;
    },
};
```

My `SdToast.vue` has this piece of code:
```vue
<template>
    <div v-for="toast in toasts" :key="toast.id">{{ toast.message }}</div>
</template>
<script>
import { ref } from '@vue/composition-api';

export default {
    setup() {
        const toasts = ref([]);

        const show = () => {};

        const hide = id => {};

        return { toasts, show, hide };
    },
};
</script>
```

And I receive this error message: `[Vue warn]: Property or method "toasts" is not defined on the instance but referenced during render. Make sure that this property is reactive, either in the data option, or for class-based components, by initializing the property. See: https://vuejs.org/v2/guide/reactivity.html#Declaring-Reactive-Properties.`

Basically if I use the component without the plugin it works fine.

**Comment by LinusBorg** [comment]
Did you make sure to install the CompositionAPI plugin before you installed your own plugin?

**Comment by MatanYadaev** [comment]
@LinusBorg Ohh.. That was the problem. Thanks!
